### ISSUE TITLE
Different usage of daily_rates in daily_spend vs cost_per_query

### ISSUE THREAD

**Issue by john-jerome** [get-select/dbt-snowflake-monitoring#75]
Hey folks!

I've noticed that in the `cost_per_query` model you have an [extra inner join](https://github.com/get-select/dbt-snowflake-monitoring/blob/main/models/cost_per_query.sql#L86) `on current_rates.is_latest_rate`, which is not the case in the `daily_spend` model. As a result, we attribute some cost to even the most recent queries, but when looking at `daily_spend` some days could be missing.

 In our case, at the time of running the models from the package today (`2023-02-02`), we didn't have a daily rate for `2023-02-01` in `SNOWFLAKE.ORGANIZATION_USAGE.RATE_SHEET_DAILY` and that means that both `2023-02-01` and `2023-02-02` are currently missing from `daily_spend`. It's not a big deal of course, but I thought it was still worth reporting.

Thanks!

**Comment by ian-whitestone** [comment]
Hey @john-jerome , thanks for reaching out. 

Your observations are correct, we added the extra join in `cost_per_query` so that model wouldn't be held up by missing rates and we would have queries from the most recent hour. We never ended up adding the same strategy to `daily_spend` because we only include records for days with complete data, and we normally have a daily rate for the last day. But in this case it seems like snowflake is delayed on populating that, so we could use that strategy there as well.

This should be rare, but nonetheless clearly possible.

I'm gonna open up a new issue to capture that potential improvement. Thanks for flagging!
